File: swarmmanagercontroller/src/main/java/co/uk/swarmbit/docker/api/nodes/NodesApi.java
package co.uk.swarmbit.docker.api.nodes;

import co.uk.swarmbit.docker.api.nodes.parameters.NodeDeleteParameters;
import co.uk.swarmbit.docker.api.nodes.parameters.NodeUpdateParameters;
import co.uk.swarmbit.docker.api.nodes.parameters.NodesListParameters;
import co.uk.swarmbit.docker.api.common.json.NodeJson;

import java.util.List;

public interface NodesApi {

    List<NodeJson> listNodes(String swarmId, NodesListParameters parameters);

    NodeJson inspectNode(String swarmId, String id);

    void updateNode(String swarmId, String id, NodeUpdateParameters parameters);

    void deleteNode(String swarmId, String id, NodeDeleteParameters parameters);

}

File: swarmmanagercontroller-new/src/main/kotlin/com/swarmbit/docker/api/nodes/NodesApi.kt
package com.swarmbit.docker.api.nodes

import com.swarmbit.docker.api.nodes.model.Node
import com.swarmbit.docker.api.nodes.parameters.NodesDeleteParameters
import com.swarmbit.docker.api.nodes.parameters.NodesListParameters
import com.swarmbit.docker.api.nodes.parameters.NodesUpdateParameters

interface NodesApi {
    fun listNodes(swarmId: String, parameters: NodesListParameters = NodesListParameters()): List<Node>

    fun inspectNode(swarmId: String, id: String): Node

    fun updateNode(swarmId: String, id: String, parameters: NodesUpdateParameters)

    fun deleteNode(swarmId: String, id: String, parameters: NodesDeleteParameters)
}
